Portal/View/Components/TextField.swift

```diff
@@ -12,60 +12,57 @@ public struct TextFieldProperties<MessageType> {
     
     public var text: String?
     public var placeholder: String?
-    public var onEvents: TextFieldEvents<MessageType>
-    
+    public var isSecureTextEntry: Bool
+    public var shouldReturn: Bool
+    public var onEvents: (TextFieldEvents) -> MessageType?
+
     fileprivate init(
         text: String? = .none,
         placeholder: String? = .none,
-        onEvents: TextFieldEvents<MessageType> = TextFieldEvents<MessageType>() ) {
+        isSecureTextEntry: Bool = false,
+        shouldReturn: Bool = false,
+        onEvents: @escaping (TextFieldEvents) -> MessageType? = { _ in return .none}) {
         self.text = text
         self.placeholder = placeholder
         self.onEvents = onEvents
+        self.isSecureTextEntry = isSecureTextEntry
+        self.shouldReturn = shouldReturn
     }
     
     public func map<NewMessageType>(
-        _ transform: (MessageType) -> NewMessageType) -> TextFieldProperties<NewMessageType> {
+        _ transform: @escaping (MessageType) -> NewMessageType) -> TextFieldProperties<NewMessageType> {
         return TextFieldProperties<NewMessageType>(
             text: self.text,
             placeholder: self.placeholder,
-            onEvents: self.onEvents.map(transform)
+            isSecureTextEntry: isSecureTextEntry,
+            onEvents: { return self.onEvents($0).map(transform) }
         )
     }
     
 }
 
-public struct TextFieldEvents<MessageType> {
+public enum TextFieldEvents {
```

Comment:
Why did you change this to an enum? What happens if a user wants to dispatch a message `onEditingBegin` and `onEditingEnd`? Would it make sense?

Reply:
Because there is a need to be sent the string of the current textField. As you cant force a messageType to containe a String, the best way is to ask for a function that receives these enums, with the respective string, and return a MessageType

Reply:
I would change `TextFieldEvents` to

```swift
struct TextFieldEvents<MessageType> {

    let onEditingBegin: ((String) -> MessageType)?
    let onEditingChanged: ((String) -> MessageType)?
    let onEditingEnd: ((String) -> MessageType)?

    fileprivate init(
        onEditingBegin: ((String) -> MessageType)? = .none,
        onEditingChanged: ((String) -> MessageType)?,
        onEditingEnd: ((String) -> MessageType)? = .none
    ) {
        self.onEditingBegin = onEditingBegin
        self.onEditingChanged = onEditingChanged
        self.onEditingEnd = onEditingEnd
    }

}
```

and then I would add a builder function in order to do something like

```swift
textFieldEvents() {
     $0.onEditingBegin = { .myCustomMessage($0) }
}
```